src/app/commands/cmd_sprite_size.cpp

```diff
@@ -39,6 +41,23 @@ namespace app {
 using namespace ui;
 using doc::algorithm::ResizeMethod;
 
+struct SpriteSizeParams : public NewParams {
+  Param<bool> ui { this, true, { "ui", "use-ui" } };
+  Param<int> width { this, 0, "width" };
+  Param<int> height { this, 0, "height" };
+  Param<bool> lockRatio { this, false, "lockRatio" };
+  Param<double> scale { this, 1.0, "scale" };
+  Param<double> scaleX { this, 1.0, "scaleX" };
+  Param<double> scaleY { this, 1.0, "scaleY" };
+  Param<ResizeMethod> method { this, ResizeMethod::RESIZE_METHOD_NEAREST_NEIGHBOR, { "method", "resize-method" } };
+
+public:
+  void setWidth(int w) { width.fromString(base::convert_to<std::string>(w).c_str()); };
+  void setHeight(int h) { height.fromString(base::convert_to<std::string>(h).c_str()); };
+  void setScaleX(double scX) { scaleX.fromString(base::convert_to<std::string>(scX).c_str()); };
+  void setScaleY(double scY) { scaleY.fromString(base::convert_to<std::string>(scY).c_str()); };
```

Comment:
These member functions `setWidth`/`setHeight`/etc. are not necessary, you can remove them because there are a way to change `Param` value (using the [`Param::operator()(const T& value)` member function](https://github.com/aseprite/aseprite/blob/4c4c20ace6334685b86d904bfe3fc0b0c50a3db5/src/app/commands/new_params.h#L94)). For example, to change the `width` field you just write:
```c++
int newValue = ...;
params.width(newValue);
```
Example in [CanvasSize](https://github.com/aseprite/aseprite/blob/4c4c20ace6334685b86d904bfe3fc0b0c50a3db5/src/app/commands/cmd_canvas_size.cpp#L342) params.

Basically, a `Param` offers a getter (`int currentValue = params.width()`) and a setter (`params.width(newValue)`).

Reply:
Got it!